automata/fa/dfa.py

```diff
@@ -602,6 +603,29 @@ def isfinite(self):
         except ValueError:
             return True
 
+    def random_word(self, k, *, seed=None):
+        self._ensure_count_for_length(k)
+        state = self.initial_state
+        if self._count_cache[k][state] == 0:
+            raise ValueError(f"Language has no words of length {k}")
+
+        result = []
+        rng = Random(seed)
+        for i in range(k):
+            cur_length = k - i
```

Comment:
I think you can fold this into the range loop, something like `for cur_length in range(k, 1, -1)`.

Reply:
Fixed